tests/components/flo/test_device.py

```diff
@@ -0,0 +1,50 @@
+"""Define tests for device-related endpoints."""
+from datetime import timedelta
+
+from homeassistant.components.flo.const import DOMAIN as FLO_DOMAIN
+from homeassistant.components.flo.device import FloDeviceDataUpdateCoordinator
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.setup import async_setup_component
+from homeassistant.util import dt
+
+from .common import TEST_PASSWORD, TEST_USER_ID
+
+from tests.common import async_fire_time_changed
+
+
+async def test_device(hass, config_entry, aioclient_mock_fixture, aioclient_mock):
+    """Test Flo by Moen device."""
+    config_entry.add_to_hass(hass)
+    assert await async_setup_component(
+        hass, FLO_DOMAIN, {CONF_USERNAME: TEST_USER_ID, CONF_PASSWORD: TEST_PASSWORD}
+    )
+    await hass.async_block_till_done()
+    assert len(hass.data[FLO_DOMAIN]["devices"]) == 1
+
+    device: FloDeviceDataUpdateCoordinator = hass.data[FLO_DOMAIN]["devices"][0]
+    assert device.api_client is not None
+    assert device.available
+    assert device.consumption_today == 3.674
+    assert device.current_flow_rate == 0
+    assert device.current_psi == 54.20000076293945
+    assert device.current_system_mode == "home"
+    assert device.target_system_mode == "home"
+    assert device.firmware_version == "6.1.1"
+    assert device.device_type == "flo_device_v2"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)